apps/frontend/src/app/5_pages/RewardsPage/components/StabilityPool/StabilityPool.tsx

```diff
@@ -60,7 +57,9 @@ export const StabilityPool: FC = () => {
   } = useGetTroves();
 
   const { checkMaintenance, States } = useMaintenance();
-  const claimLocked = checkMaintenance(States.ZERO_STABILITY_CLAIM);
+  const rewardsLocked = checkMaintenance(States.REWARDS_FULL);
+  const stabilityClaimLocked = checkMaintenance(States.ZERO_STABILITY_CLAIM);
+  const claimLocked = stabilityClaimLocked || rewardsLocked;
```

Comment:
```suggestion
  const claimLocked = useMemo(
    () =>
      checkMaintenance(States.REWARDS_FULL) ||
      checkMaintenance(States.ZERO_STABILITY_CLAIM),
    [States.REWARDS_FULL, States.ZERO_STABILITY_CLAIM, checkMaintenance],
  );
```
You don't use `rewardsLocked` and `stabilityClaimLocked` anywhere except `claimLocked`. I don't mind them for the readability so you can keep them if you want but at least use `useMemo` in `claimLocked`.